### PR TITLE
socket-util: drop obsolete FIXME comment and add test

### PR DESCRIPTION
**Pull request by keszybz**
We are pretty careful to reject abstract sockets that are too long to fit in
the address structure as a NUL-terminated string. And since we parse sockets
as strings, it is not possible to embed a NUL in the the address either.
So drop the comment, and add a test to verify that we get the length-check
right.

This should make LGTM happier.

### CODE REVIEW

File: src/basic/socket-util.c [hunk 1]
```diff
@@ -648,12 +648,7 @@ int sockaddr_pretty(const struct sockaddr *_sa, socklen_t salen, bool translate_
                                 return -ENOMEM;
 
                 } else if (sa->un.sun_path[0] == 0) {
-                        /* abstract */
-
-                        /* FIXME: We assume we can print the
-                         * socket path here and that it hasn't
-                         * more than one NUL byte. That is
-                         * actually an invalid assumption */
+                        /* Abstract socket */
```
Comment (poettering) [inline src/basic/socket-util.c]: well, we make no restrictions on how the "struct sockaddr" came to be. While we are doing this properly it doesn't mean everybody does. 

I would feel more comfortable if we'd simply generate an error if there's an embedded NUL in this case then simply assume it doesn't happen. i.e. return EADDRNOTAVAIL in if there's any NUL char in sun_path that is not at the end?

**Comment by evverx** [comment]
This pull request **fixes 1 alert** when merging 36f3c3a9f869f01659e24cfdf43f7553d8da9326 into b2ac2b01c8ddaabbe63bcb7168dd752f44320d86 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-2fe11a78214c2fe6dc150ac2fa0007abbfa7ddb7)

**fixed alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by keszybz** [comment]
What a rabbit-hole to go down into ;) I reworked the patch significantly, ptal.

File: src/test/test-socket-util.c [hunk 2]
```diff
@@ -42,61 +44,63 @@ static void test_ifname_valid(void) {
...
-        assert_se(socket_address_parse(&a, "[::1]8888") < 0);
-        assert_se(socket_address_parse(&a, "::1") < 0);
-        assert_se(socket_address_parse(&a, "[::1]:0") < 0);
-        assert_se(socket_address_parse(&a, "[::1]:65536") < 0);
-        assert_se(socket_address_parse(&a, "[a:b:1]:8888") < 0);
-
-        assert_se(socket_address_parse(&a, "8888") >= 0);
-        assert_se(a.sockaddr.sa.sa_family == (socket_ipv6_is_supported() ? AF_INET6 : AF_INET));
+        r = socket_address_parse(&a, in);
+        log_info("\"%s\" → %s", in, r >= 0 ? "✓" : "✗");
```
Comment (poettering) [inline src/test/test-socket-util.c]: we should have a helper call for the checkmark/crossmark thing similar to yes_no(), it's just too beautiful ;-)

**Comment by poettering** [comment]
lgtm

**Comment by evverx** [comment]
This pull request **fixes 1 alert** when merging 15dca3711da63ac5fcfa01187a3b964d7042b99c into b806f0b0aa043656470dab27bb9d1bd6015e1948 - [view on LGTM.com](https://lgtm.com/projects/g/systemd/systemd/rev/pr-e67c9b0be2ae2ef86262e7cafd17b486cb7e32c6)

**fixed alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**